apps/src/turtle/blocks.js

```diff
@@ -259,35 +259,84 @@ exports.install = function (blockly, blockInstallOptions) {
     }
   };
 
-blockly.Blocks.point_to = {
-    // Block for pointing to a specified direction
-    helpUrl: '',
-    init: function () {
-      this.setHSV(184, 1.00, 0.74);
-      this.appendDummyInput()
-          .appendTitle(msg.pointTo());
-      this.appendDummyInput()
-          .appendTitle(new blockly.FieldTextInput('0', blockly.FieldTextInput.numberValidator), 'DIRECTION')
+  generator.draw_turn_inline = function () {
+    // Generate JavaScript for turning left or right.
+    var value = window.parseFloat(this.getTitleValue('VALUE'));
+    return 'Turtle.' + this.getTitleValue('DIR') +
+        '(' + value + ', \'block_id_' + this.id + '\');\n';
+  };
+
+  function createPointToDirectionBlock(blockName) {
+    return {
+      helpUrl: '',
+      init: function () {
+        this.setHSV(184, 1.00, 0.74);
+        this.appendDummyInput()
+            .appendTitle(msg.pointTo());
+        this.setPreviousStatement(true);
+        this.setInputsInline(true);
+        this.setNextStatement(true);
+        this.setTooltip(msg.pointTo());
+
+        appendToPointToDirectionBlock(blockName, this);
+      }
+    };
+  }
+
+  function appendToPointToDirectionBlock(blockName, block) {
```

Comment:
Rather than creating a method here which is intended to be used in one of three different ways, you could have the `createPointToDirectionBlock` accept an optional method to handle the extra initialization step. Something like this:

```javascript
function createPointToDirectionBlock(onInit) {
  return {
    helpUrl: '',
    init: function () {
      this.setHSV(184, 1.00, 0.74);
      this.appendDummyInput()
          .appendTitle(msg.pointTo());
      this.setPreviousStatement(true);
      this.setInputsInline(true);
      this.setNextStatement(true);
      this.setTooltip(msg.pointTo());
      if (onInit) {
        onInit(this);
      }
    }
  };
}


blockly.Blocks.point_to_direction = createPointToDirectionBlock(function (block) {
  block.appendDummyInput()
      .appendTitle(new blockly.FieldAngleTextInput('VALUE', '0'),
          'DIRECTION')
      .appendTitle(msg.degrees());
});
```

Reply:
Whoa.  Callbacks it is.